sc-kpm/sc-agents-common/utils/CommonUtils.cpp

```diff
@@ -42,12 +46,21 @@ int CommonUtils::readInt(ScMemoryContext * ms_context, const ScAddr & scLink)
 
 int CommonUtils::readNumber(ScMemoryContext * ms_context, const ScAddr & number)
 {
-  ScAddr scLink = IteratorUtils::getFirstByOutRelation(ms_context, number, CoreKeynodes::nrel_idtf);
+  SC_CHECK_PARAM(number, ("Invalid number node address"))
+
+  ScAddr scLink = IteratorUtils::getAnyByOutRelation(ms_context, number, CoreKeynodes::nrel_idtf);
   return readInt(ms_context, scLink);
 }
 
 string CommonUtils::readString(ScMemoryContext * ms_context, const ScAddr & scLink)
 {
+  return getLinkContent(ms_context, scLink);
+}
+
+string CommonUtils::getLinkContent(ScMemoryContext * ms_context, const ScAddr & scLink)
```

Comment:
Link content is not string only

Reply:
A link content represents as a string stream. When getting link content it is formatted to another type

Reply:
The method provides safe link content getting